### PR TITLE
feat: workspace cli controls for notebooks

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

DET-8698

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/cli/workspace.py [hunk 1]
```diff
@@ -22,6 +22,29 @@
     "Agent Group",
 ]
 
+workspace_arg: Arg = Arg("--workspace-name", type=str, help="workspace name")
+
+
+def get_workspace_names(session: api.Session) -> Dict[int, str]:
```
Comment (hamidzr) [inline harness/determined/cli/workspace.py:28]: we might decide to do this translation in master at the api level which would move these checks there.

File: harness/determined/cli/notebook.py [hunk 2]
```diff
@@ -13,13 +15,48 @@
 from determined.common.declarative_argparse import Arg, Cmd, Group
 
 
+def expected_exceptions(func: Callable[[argparse.Namespace], Any]) -> Callable[..., Any]:
+    def wrapper(args: argparse.Namespace) -> Any:
+        try:
+            return func(args)
+        except Exception as e:
+            """
+            DISCUSS: is this a good pattern in Python? if so we could do this at argparse.Cmd
```
Comment (hamidzr) [inline harness/determined/cli/notebook.py:24]: thoughts?

**Review by eecsliu — COMMENTED** [review]
LGTM, except error handling - should be more specific and we should talk to engineering if we want to implement that more broadly.

File: harness/determined/cli/notebook.py (hunk 2 above)
Comment (hamidzr) [reply harness/determined/cli/notebook.py]: I separated this into a different branch https://github.com/determined-ai/determined/compare/master...hamidzr:determined:cli-err-handling?expand=1 and I'll follow up on it separately. switched this to a more local solution

**Comment by hamidzr** [comment]
thanks for the review

**Pull request merged**